internal/controllers/machine/machine_controller_phases_test.go

```diff
@@ -1003,82 +1041,211 @@ func TestReconcileInfrastructure(t *testing.T) {
 					},
 				},
 			},
-			expectResult:  ctrl.Result{},
-			expectError:   false,
-			expectChanged: true,
+			InfraMachineGetError: nil,
+			expectResult:         ctrl.Result{},
+			expectError:          false,
+			expected: func(g *WithT, m *clusterv1.Machine) {
+				g.Expect(m.Status.InfrastructureReady).To(BeTrue())
+				g.Expect(ptr.Deref(m.Spec.ProviderID, "")).To(Equal("test://id-1"))
+				g.Expect(m.Spec.FailureDomain).To(BeNil())
+				g.Expect(m.Status.Addresses).ToNot(BeNil())
+			},
+		},
+		{
+			name:    "infra machine ready and with all the optional fields, it should reconcile and data should surface on the machine",
+			machine: defaultMachine.DeepCopy(),
+			infraMachine: map[string]interface{}{
+				"kind":       "GenericInfrastructureMachine",
+				"apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
+				"metadata": map[string]interface{}{
+					"name":      "infra-config1",
+					"namespace": metav1.NamespaceDefault,
+				},
+				"spec": map[string]interface{}{
+					"providerID":    "test://id-1",
+					"failureDomain": "foo",
+				},
+				"status": map[string]interface{}{
+					"ready": true,
+					"addresses": []interface{}{
+						map[string]interface{}{
+							"type":    "InternalIP",
+							"address": "10.0.0.1",
+						},
+						map[string]interface{}{
+							"type":    "InternalIP",
+							"address": "10.0.0.2",
+						},
+					},
+				},
+			},
+			InfraMachineGetError: nil,
+			expectResult:         ctrl.Result{},
+			expectError:          false,
 			expected: func(g *WithT, m *clusterv1.Machine) {
 				g.Expect(m.Status.InfrastructureReady).To(BeTrue())
-				g.Expect(m.GetOwnerReferences()).NotTo(ContainRefOfGroupKind("cluster.x-k8s.io", "MachineSet"))
+				g.Expect(ptr.Deref(m.Spec.ProviderID, "")).To(Equal("test://id-1"))
+				g.Expect(ptr.Deref(m.Spec.FailureDomain, "")).To(Equal("foo"))
+				g.Expect(m.Status.Addresses).ToNot(BeNil())
 			},
 		},
 		{
-			name: "ready bootstrap, infra, and nodeRef, machine is running, infra object is deleted, expect failed",
+			name:    "infra machine ready but paused, it should reconcile but no data should surface on the machine",
```

Comment:
```suggestion
			name:    "infra machine ready but paused, it should reconcile and data should surface on the machine",
```

I would also change "but paused" to "and paused", "but paused" seems to imply that it has any sort of effect (same for the bootstrap test case above)

Reply:
Fixed!